proto/proto-backplane-grpc-util/src/main/java/io/deephaven/grpc_api/util/OperationHelper.java

```diff
@@ -0,0 +1,76 @@
+        }
+        switch (opCase) {
+            case EMPTY_TABLE:
+            case TIME_TABLE:
+                return Stream.empty();
+            case DROP_COLUMNS:
+                return Stream.of(op.getDropColumns().getSourceId());
+            case UPDATE:
+                return Stream.of(op.getUpdate().getSourceId());
+            case LAZY_UPDATE:
+                return Stream.of(op.getLazyUpdate().getSourceId());
+            case VIEW:
+                return Stream.of(op.getView().getSourceId());
+            case UPDATE_VIEW:
+                return Stream.of(op.getUpdateView().getSourceId());
+            case SELECT:
+                return Stream.of(op.getSelect().getSourceId());
+            case SELECT_DISTINCT:
+                return Stream.of(op.getSelectDistinct().getSourceId());
+            case FILTER:
+                return Stream.of(op.getFilter().getSourceId());
+            case UNSTRUCTURED_FILTER:
+                return Stream.of(op.getUnstructuredFilter().getSourceId());
+            case SORT:
+                return Stream.of(op.getSort().getSourceId());
+            case HEAD:
+                return Stream.of(op.getHead().getSourceId());
+            case TAIL:
+                return Stream.of(op.getTail().getSourceId());
+            case HEAD_BY:
+                return Stream.of(op.getHeadBy().getSourceId());
+            case TAIL_BY:
+                return Stream.of(op.getTailBy().getSourceId());
+            case UNGROUP:
+                return Stream.of(op.getUngroup().getSourceId());
+            case MERGE:
+                return op.getMerge().getSourceIdsList().stream();
+            case CROSS_JOIN:
+                return Stream.of(op.getCrossJoin().getLeftId(), op.getCrossJoin().getRightId());
+            case NATURAL_JOIN:
+                return Stream.of(op.getNaturalJoin().getLeftId(), op.getNaturalJoin().getRightId());
+            case EXACT_JOIN:
+                return Stream.of(op.getExactJoin().getLeftId(), op.getExactJoin().getRightId());
+            case LEFT_JOIN:
+                return Stream.of(op.getLeftJoin().getLeftId(), op.getLeftJoin().getRightId());
+            case AS_OF_JOIN:
+                return Stream.of(op.getAsOfJoin().getLeftId(), op.getAsOfJoin().getRightId());
+            case COMBO_AGGREGATE:
+                return Stream.of(op.getComboAggregate().getSourceId());
+            case SNAPSHOT:
+                return Stream.of(op.getSnapshot().getLeftId(), op.getSnapshot().getRightId());
+            case FLATTEN:
+                return Stream.of(op.getFlatten().getSourceId());
+            case RUN_CHART_DOWNSAMPLE:
+                return Stream.of(op.getRunChartDownsample().getSourceId());
+            case OP_NOT_SET:
+                throw new IllegalStateException("Operation id not set");
+            default:
+                throw new IllegalStateException("Operation not implemented yet, " + opCase);
```

Comment:
I'm not sure if there's any good way to construct all `Operation`s based on OpCase enum - the only way to create an Operation is via `Operation.newBuilder().setX(...).build()`. We could add some side-by-side utilities that essentially parses out Operation into its respective components into strongly typed visitor if we wanted - but we still have the original base-case that the multiplexer code needs to be complete, but we don't have a good way to test it against all Operations afaik.

(It's a bit unfortunate that protobuf doesn't generate this type of code for us. But, it may be a useful pattern that would make consuming our protobufs easier...)

Reply:
Added.